pkg/app/client/client.go

```diff
@@ -551,20 +555,28 @@ func (c *Client) CloseIdleConnections() {
 }
 
 func (c *Client) mCleaner() {
+	c.cleaner(c.m)
+}
+
+func (c *Client) msCleaner() {
+	c.cleaner(c.ms)
+}
+
+func (c *Client) cleaner(m map[string]client.HostClient) {
 	for {
 		time.Sleep(10 * time.Second)
-		if c.mClean() {
+		if c.cleanHostClients(m) {
 			break
 		}
 	}
 }
 
-func (c *Client) mClean() bool {
```

Comment:
any issues if we always clean two maps in this func?

Reply:
I think that cleaning both maps itself is not an issue, but since only one of them is actually used depending on the value of `isTLS`, I think cleaning just one of them should be sufficient. What do you think?

Reply:
the most critical problem is you're reading map pointers without lock.

Reply:
omg I'll fix it thank you

Reply:
@xiaost fixed. please take a look again.

Reply:
@xiaost Fixed. Please take a look again.